Source/SteveSloth/Private/MyEnemyBaseClass.cpp

```diff
@@ -76,9 +83,38 @@ void AMyEnemyBaseClass::Tick(float DeltaTime)
 		GetWorldTimerManager().SetTimer(DespawnTimerHandle, this, &AMyEnemyBaseClass::Despawn, DESPAWN_TIMER_AMOUNT, false);
 		IsDead = true;
 	}
+	//Check if frozen
+	else if (IsFrozen)
+	{
+		//show visual of freezing here
+
+		IsFrozen = true;
+		IsIdle = true;
+		IsChasing = true;
+		IsPatroling = true;
+		IsAttackingMelee = true;
+		IsAttackingRanged = true;
+
```

Comment:
Just wondering why all these are set to true in the frozen check?

Reply:
for some reason my other comment didn't save, it was pretty much the same as the previous comment just why on are they all being set back to false on lines 114-119, I understand IsFrozen but why the other ones?

Reply:
this is because of the state machine, the enemies will never go into a state if their respective bool is true. so it will never go into any state. however there is something I need to add

Reply:
Ah okay, thank you for clarifying